internal/provisioner/kops_provisioner_cluster_installation.go

```diff
@@ -580,15 +614,37 @@ func (provisioner *KopsProvisioner) prepareCILicenseSecret(installation *model.I
 	return licenseSecretName, nil
 }
 
-func (provisioner *KopsProvisioner) deleteLicenseSecret(clusterInstallation *model.ClusterInstallation, k8sClient *k8s.KubeClient, logger log.FieldLogger) error {
-	secretName := fmt.Sprintf("%s-license", makeClusterInstallationName(clusterInstallation))
-	err := k8sClient.Clientset.CoreV1().Secrets(clusterInstallation.Namespace).Delete(context.Background(), secretName, metav1.DeleteOptions{})
-	if k8sErrors.IsNotFound(err) {
-		logger.Warnf("Secret %s/%s not found, assuming already deleted", clusterInstallation.Namespace, secretName)
-		return nil
+func generateCILicenseName(installation *model.Installation, clusterInstallation *model.ClusterInstallation) string {
+	sum := fmt.Sprintf("%x", sha256.Sum256([]byte(installation.License)))
+	if len(sum) > 7 {
+		sum = sum[0:6]
 	}
+
+	return fmt.Sprintf("%s-%s-license", makeClusterInstallationName(clusterInstallation), sum)
+}
+
+// cleanupOldLicenseSecrets removes an secrets matching the license naming
+// convention except the current license secret name. Pass in a blank name value
+// to cleanup all license secrets.
+func cleanupOldLicenseSecrets(currentSecretName string, clusterInstallation *model.ClusterInstallation, k8sClient *k8s.KubeClient, logger log.FieldLogger) error {
+	secrets, err := k8sClient.Clientset.CoreV1().Secrets(clusterInstallation.Namespace).List(context.Background(), metav1.ListOptions{})
```

Comment:
I think we could actually use labels to mark secrets with the license, which might be a little cleaner than relying on naming but this should work fine either so 1/5 on that.

Reply:
Yeah, agreed. I had to start with this simply because the original license secrets didn't have any labels, but we can definitely move to this in a follow-up.